File: kratos.client/src/models/Empresas.js
class Empresas {
    constructor({
        id = undefined, // Cambiado de null a undefined para que no se incluya en el JSON
        contrasena = '',
        confirmarContrasena = '',
        tiposociedadId = null,
        actividadId = null,
        regimenId = null,
        token = '',
        razonSocial = '',
        nombreComercial = '',
        nit = '',
        dv = '',
        telefono = '',
        email = '',
        representanteLegal = '',
        activo = true,
        creadoEn = new Date().toISOString(),
        actualizadoEn = new Date().toISOString()
    } = {}) {
        this.id = id;
        this.contrasena = contrasena;
        this.confirmarContrasena = confirmarContrasena;
        this.tiposociedadId = tiposociedadId;
        this.actividadId = actividadId;
        this.regimenId = regimenId;
        this.token = token;
        this.razonSocial = razonSocial;
        this.nombreComercial = nombreComercial;
        this.nit = nit;
        this.dv = dv;
        this.telefono = telefono;
        this.email = email;
        this.representanteLegal = representanteLegal;
        this.activo = activo;
        this.creadoEn = creadoEn;
        this.actualizadoEn = actualizadoEn;
    }

    validate() {
        const errors = [];

        // Validacion de contraseñas
        if (!this.contrasena || this.contrasena.trim() === '') {
            errors.push('El campo contraseña es obligatorio');
        }
        if (!this.confirmarContrasena || this.confirmarContrasena.trim() === '') {
            errors.push('El campo confirmar contraseña es obligatorio');
        }
        if (this.contrasena && this.confirmarContrasena && this.contrasena !== this.confirmarContrasena) {
            errors.push('La contraseña y la confirmacion no coinciden');
        }
        if (this.contrasena && this.contrasena.length > 100) {
            errors.push('La contraseña no puede exceder los 100 caracteres');
        }

        // Validacion de campos obligatorios
        const requiredFields = {
            razonSocial: 'Razon Social',
            email: 'Correo Electronico',
            nit: 'NIT',
            telefono: 'Telefono',
            token: 'Token',
            nombreComercial: 'Nombre Comercial',
            representanteLegal: 'Representante Legal',
            tiposociedadId: 'Tipo de Sociedad',
            actividadId: 'Actividad Economica',
            regimenId: 'Regimen Tributario'
        };

        Object.entries(requiredFields).forEach(([field, name]) => {
            if (!this[field] && this[field] !== 0) {
                errors.push(`El campo ${name} es obligatorio`);
            }
        });

        // Validacion de longitud maxima
        const maxLengthFields = {
            razonSocial: 100,
            email: 100,
            nit: 100,
            telefono: 100,
            token: 100,
            nombreComercial: 100,
            representanteLegal: 100,
            dv: 100
        };

        Object.entries(maxLengthFields).forEach(([field, max]) => {
            if (this[field] && this[field].length > max) {
                errors.push(`El campo ${field} no puede exceder los ${max} caracteres`);
            }
        });

        if (errors.length > 0) {
            throw new Error(errors.join(', '));
        }
    }

    toJSON() {
        const jsonData = {
            razonSocial: this.razonSocial,
            nombreComercial: this.nombreComercial,
            nit: this.nit,
            dv: this.dv,
            telefono: this.telefono,
            email: this.email,
            representanteLegal: this.representanteLegal,
            tiposociedadId: Number(this.tiposociedadId),
            actividadId: Number(this.actividadId),
            regimenId: Number(this.regimenId),
            token: this.token,
            contrasena: this.contrasena,
            confirmarContrasena: this.confirmarContrasena,
            activo: this.activo,
            creadoEn: this.creadoEn,
            actualizadoEn: this.actualizadoEn
        };

        // Solo incluir el ID si esta definido
        if (this.id !== undefined) {
            jsonData.id = this.id;
        }

        return jsonData;
    }

    static fromFormData(formData) {
        return new Empresas({
            ...formData,
            activo: true,
            creadoEn: new Date().toISOString(),
            actualizadoEn: new Date().toISOString()
        });
    }
}

export default Empresas;

File: Kratos.Server/Models/Kratos/Empresas.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;


namespace kratos.Server.Models.Kratos
{

    [Table("Empresas")] // Especifica el nombre de la tabla en MySQL
    public class Empresas
    {
        [Key]
        [Column("id")]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int id { get; set; }

        [Required(ErrorMessage = "El campo Contraseña es obligatorio")]
        [MaxLength(100, ErrorMessage = "La contraseña no puede exceder 100 caracteres")]
        [DataType(DataType.Password)]
        [Column("contrasena", TypeName = "varchar(100)")]
        public string contrasena { get; set; }


        [Required(ErrorMessage = "El campo Confirmar Contraseña es obligatorio")]
        [MaxLength(100, ErrorMessage = "La confirmación no puede exceder 100 caracteres")]
        [DataType(DataType.Password)]
        [Compare("contrasena", ErrorMessage = "Las contraseñas no coinciden")]
        public string confirmarContrasena { get; set; }

        [Required(ErrorMessage = "El campo Tipo Sociedad es obligatorio")]
        [Column("tiposociedadId")]
        [ForeignKey("empresasociedadFk")]
        public int tiposociedadId { get; set; }

    
        public virtual TiposSociedades? empresasociedadFk { get; set; }

        [Required(ErrorMessage = "El campo Actividad Económica es obligatorio")]
        [Column("actividadId")]
        [ForeignKey("empresaactividadFk")]
        public int actividadId { get; set; }

      
        public virtual ActividadEconomicas? empresaactividadFk { get; set; }

        [Required(ErrorMessage = "El campo Régimen Tributario es obligatorio")]
        [Column("regimenId")]
        [ForeignKey("empresaregimenFk")]
        public int regimenId { get; set; }

        public virtual RegimenesTributarios? empresaregimenFk { get; set; }

        [Required(ErrorMessage = "El campo Token es obligatorio")]
        [MaxLength(100, ErrorMessage = "El campo Token debe tener máximo 100 caracteres")]
        [Column("token", TypeName = "varchar(100)")]
        public string token { get; set; }

        [Required(ErrorMessage = "El campo Razón Social es obligatorio")]
        [MaxLength(100, ErrorMessage = "El campo Razón Social debe tener máximo 100 caracteres")]
        [Column("razonSocial", TypeName = "varchar(100)")]
        public string razonSocial { get; set; }

        [Required(ErrorMessage = "El campo Nombre Comercial es obligatorio")]
        [MaxLength(100, ErrorMessage = "El campo Nombre Comercial debe tener máximo 100 caracteres")]
        [Column("nombreComercial", TypeName = "varchar(100)")]
        public string nombreComercial { get; set; }

        [Required(ErrorMessage = "El campo NIT es obligatorio")]
        [MaxLength(100, ErrorMessage = "El campo NIT debe tener máximo 100 caracteres")]
        [Column("nit", TypeName = "varchar(100)")]
        public string nit { get; set; }

        [MaxLength(100, ErrorMessage = "El campo DV debe tener máximo 100 caracteres")]
        [Column("dv", TypeName = "varchar(100)")]
        public string dv { get; set; }

        [Required(ErrorMessage = "El campo Teléfono es obligatorio")]
        [DataType(DataType.PhoneNumber)]
        [Column("telefono", TypeName = "varchar(20)")]
        public string telefono { get; set; }

        [Required(ErrorMessage = "El campo Email es obligatorio")]
        [MaxLength(100, ErrorMessage = "El campo Email debe tener máximo 100 caracteres")]
        [DataType(DataType.EmailAddress)]
        [Column("email", TypeName = "varchar(100)")]
        public string email { get; set; }

        [Required(ErrorMessage = "El campo Representante Legal es obligatorio")]
        [MaxLength(100, ErrorMessage = "El campo Representante Legal debe tener máximo 100 caracteres")]
        [Column("representanteLegal", TypeName = "varchar(100)")]
        public string representanteLegal { get; set; }

        [Column("activo", TypeName = "tinyint(1)")]
        public bool activo { get; set; } = true; // Valor por defecto

        [DataType(DataType.Date)]
        [Column("creadoEn", TypeName = "date")]
        public DateTime creadoEn { get; set; } = DateTime.UtcNow; // Valor por defecto

        [DataType(DataType.Date)]
        [Column("actualizadoEn", TypeName = "date")]
        public DateTime actualizadoEn { get; set; } = DateTime.UtcNow; // Valor por defecto
    }
}